python/tvm/_ffi/runtime_ctypes.py

```diff
@@ -96,6 +98,9 @@ def __init__(self, type_str):
             self.type_code = DataTypeCode.HANDLE
             bits = 64
             head = ""
+        elif head.startswith("bfloat"):
+            self.type_code = 4
```

Comment:
not sure if it is good to hard code here

Reply:
> not sure if it is good to hard code here

Change to DataTypeCode. TVM refactors a lot (which is good). And when this PR is raised, all the type code here uses hard codes.

The other two issues you raised are also changed as required.